oap-server/server-receiver-plugin/aws-firehose-receiver/src/main/java/org/apache/skywalking/oap/server/receiver/aws/firehose/FirehoseHTTPHandler.java

```diff
@@ -0,0 +1,86 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.skywalking.oap.server.receiver.aws.firehose;
+
+import com.linecorp.armeria.server.annotation.ConsumesJson;
+import com.linecorp.armeria.server.annotation.Post;
+import com.linecorp.armeria.server.annotation.ProducesJson;
+import io.grpc.stub.StreamObserver;
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceRequest;
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceResponse;
+import io.opentelemetry.proto.metrics.v1.ResourceMetrics;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Base64;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.skywalking.oap.server.receiver.otel.otlp.OpenTelemetryMetricHandlerExporter;
+
+@Slf4j
+public class FirehoseHTTPHandler {
+
+    private final OpenTelemetryMetricHandlerExporter openTelemetryMetricHandlerExporter;
+
+    private static final StreamObserver<ExportMetricsServiceResponse> STREAM_OBSERVER = new StreamObserver<ExportMetricsServiceResponse>() {
+        @Override
+        public void onNext(final ExportMetricsServiceResponse exportMetricsServiceResponse) {
+        }
+
+        @Override
+        public void onError(final Throwable throwable) {
+        }
+
+        @Override
+        public void onCompleted() {
+        }
+    };
+
+    public FirehoseHTTPHandler(final OpenTelemetryMetricHandlerExporter openTelemetryMetricHandlerExporter) {
+        this.openTelemetryMetricHandlerExporter = openTelemetryMetricHandlerExporter;
+    }
+
+    @Post("/aws/firehose/metrics")
+    @ConsumesJson
+    @ProducesJson
+    public FirehouseRes collectMetrics(final FirehouseReq firehouseReq) {
+        final ExportMetricsServiceRequest.Builder builder = ExportMetricsServiceRequest.newBuilder();
+        try {
+            for (RequestData record : firehouseReq.getRecords()) {
+                final ByteArrayInputStream byteInputStream = new ByteArrayInputStream(
+                    Base64.getDecoder().decode(record.getData()));
+                skipVarIntBytes(byteInputStream);
+                builder.addResourceMetrics(ResourceMetrics.parseFrom(byteInputStream));
+            }
+        } catch (IOException e) {
+            log.warn("Only OpenTelemetry format is accepted", e);
+            return new FirehouseRes(firehouseReq.getRequestId(), System.currentTimeMillis(),
+                                    "Only OpenTelemetry format is accepted");
+        }
+        openTelemetryMetricHandlerExporter.export(builder.build(), STREAM_OBSERVER);
+        return new FirehouseRes(firehouseReq.getRequestId(), System.currentTimeMillis(), null);
+
+    }
+
+    private static void skipVarIntBytes(final ByteArrayInputStream byteInputStream) {
```

Comment:
Why is there a method for skipping `VarInt`? This seems strange.

Reply:
The record starts with `VarInt ` to indicate the data length in bytes, so I skip the bytes which are `VarInt`, the remaining bytes could be parsed to `ResourceMetrics`

Reply:
<img width="891" alt="image" src="https://user-images.githubusercontent.com/5441976/213861470-19bfb853-8801-44d0-ab5e-72b77af6c2f7.png">

According to this, we should
1. Read the length as N
2. Decode the following N bytes from the stream
3. Back to <1> if the stream is not ended. 

But currently, the implementation seems to assume there is only one ExportMetricsServiceRequest per stream, is this correct?

Reply:
Yep, I'm wrong.